packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart

```diff
@@ -133,17 +144,28 @@ class LoadBundleTask
   ///Tracks progress of loadBundle snapshots as the documents are loaded into cache
   Stream<LoadBundleTaskProgress> stream() {
     late StreamController<LoadBundleTaskProgress> controller;
-    controller =
-        StreamController<LoadBundleTaskProgress>.broadcast(onListen: () {
+    controller = StreamController<LoadBundleTaskProgress>(onListen: () {
       /// Calls underlying onProgress method on a LoadBundleTask [jsObject].
       jsObject.onProgress(
           allowInterop((firestore_interop.LoadBundleTaskProgressJsImpl data) {
-        controller.add(LoadBundleTaskProgress._fromJsObject(data));
+        LoadBundleTaskProgress taskProgress =
+            LoadBundleTaskProgress._fromJsObject(data);
+
+        if (LoadBundleTaskState.error != taskProgress.taskState) {
+          //error handled in addError() call below.
+          controller.add(taskProgress);
+        }
+      }));
+
+      jsObject.then(allowInterop((value) {
+        controller.close();
```

Comment:
Don't we need to handle "value" here?

Reply:
Already handled in the `onProgress` [event handler](https://github.com/FirebaseExtended/flutterfire/blob/00f0ab6f0c6b29c96e9664e0d5939ba5c7bb033f/packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart#L156).

Reply:
Are you sure that `.then` won't emit something _after_ `onProgress`

Like onProgress sending the 99% completion then .then sending the 100%

Reply:
positive.

Reply:
In that case, lgtm